cmd/migrator/app/migrator.go

```diff
@@ -77,6 +89,65 @@ func Run(ctx context.Context) error {
 		dynamic,
 		migration,
 	)
-	c.Run(ctx)
+
+	kubeClient, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+
+	leaderElectionConfig := newLeaderElectionConfig(kubeClient.CoreV1())
+
+	leaderElectionConfig.Callbacks.OnStartedLeading = func(ctx context.Context) {
+		c.Run(ctx)
+	}
+
+	leaderelection.RunOrDie(ctx, *leaderElectionConfig)
+
 	panic("unreachable")
+
+}
+
+func getDefaultResourceLockNamespace() string {
+	if data, err := ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace"); err == nil {
+		if ns := strings.TrimSpace(string(data)); len(ns) > 0 {
+			return ns
+		}
+	}
+	return "default"
```

Comment:
i would maybe panic here if we missing the namespace file... but even better I would pass the namespace as flag?